python/ray/autoscaler/_private/aws/config.py

```diff
@@ -415,6 +415,12 @@ def _configure_subnet(config):
                 "to populate the list of subnets and trying this again.".
                 format(config["provider"]["availability_zone"]))
 
+    # Generate subnets in only one VPC. This is to make sure we don't attempt
+    # to launch a node into a VPC outside the security group
+    # generated by _configure_security_group.
+    # See https://github.com/ray-project/ray/pull/14868.
+    subnets = [s for s in subnets if s.vpc_id == subnets[0].vpc_id]
```

Comment:
Does it make sense to combine this line with the line below?
i.e.
`subnet_ids = [s.subnet_id for s in subnets if s.vpc_id == subnets[0].vpc_id]`

Reply:
good idea! that's cleaner.